UI/src/view/defaultLocationButton.js

```diff
@@ -1,23 +1,62 @@
 (function (root) {
-  root.DefaultLocationButton = function (map, container, backend) {
+  root.DefaultLocationButton = function (map, container, backend, assetTypeConfig, defaultAssetType, defaultCoordinates) {
+    var vkm = function() {
+      if(zoom <= 5){
+        backend.getMunicipalityFromCoordinates(lon, lat, function (vkmResult) {
+              var municipalityInfo = vkmResult.kunta ? vkmResult.kunta : "Suomi";
+              municipality = municipalityInfo;
+              ely = vkmResult.ely_nimi ? vkmResult.ely_nimi : "Suomi";
+            }, function () {
+              municipality = 'Suomi';
+              ely = 'Suomi';
+            }
+        );
+      }
+    };
+
+    var defaultAsset = defaultAssetType;
+    var defaultLocation = defaultCoordinates;
+    var municipality = 'Suomi';
+    var ely = 'Suomi';
+    var lon = defaultLocation.shift();
+    var lat = defaultLocation.shift();
+    var zoom = defaultLocation.shift();
+    var roles;
+    var places;
+
     var element =
       '<div class="default-location-btn-container"></div>';
 
     container.append(element);
 
 
     eventbus.on('roles:fetched', function(userInfo) {
-      if (!(_.some(userInfo.roles, function(role) {return role === "viewer"; })))
-        $('.default-location-btn-container').append('<button class="btn btn-sm btn-tertiary" id="default-location-btn">Muuta oletussijainniksi</button>');
+      if (!(_.some(userInfo.roles, function(role) {return (role === "viewer" || role === "serviceRoadMaintainer");}))) {
+          roles = userInfo.roles[0];
+          if (roles === "busStopMaintainer") {
+              places = backend.getUserElyConfiguration();
+              municipality = ely;
+          } else if (roles === "operator") {
+              places = [];
+          } else {
+              places = backend.getUnverifiedMunicipalities();
```

Comment:
unrelated to this user story: function could be renamed to getMunicpalities or something. It only fetches the municipality names and ids, nothing to do with unverified.

Reply:
I agree. I will change that.